### DISCUSSION TITLE
How to store/download steam sdk to compile

### DISCUSSION THREAD

**Discussion by Armaldio** [greenworksjs/greenworks#3] [Q&A]
I'd like not to store the steam sdk in the repo for multiple reasons:
- License, probably
- Have to be updated everytime a new version is out
- Take space in the repo
- Not part of the source itself

In the previous repository, I zipped the sdk and added a password. But that's not ideal.
Does anyone know a way of
- Downloading the sdk programmatically from steam
- Hosting safely and be able to download without restrictions
- Bonus if the URL is always the same, event if the sdk update

**Comment by Gurigraphics** [comment]
> * Hosting safely and be able to download without restrictions

I think it's unnecessary. It is already hosted on Steam. 

Can be faster to access the link direct than to execute commands
https://partner.steamgames.com/?goto=%2Fdownloads%2Fsteamworks_sdk_153a.zip

By code, f you are not logged into partner.steamgames.com, it will give an error.

Then, as you need login, I think I would not use code

If you are going to do this with code, need add the information that you first have to login.

The last sdk link can be updated in a txt or URL
```
sdk_link.txt
```
```
https://partner.steamgames.com/downloads/steamworks_sdk_153a.zip
```

Then you can download with node/curl

Create a file
```
download.js
```
Edit the output folder
```js
const cp = require('child_process');
var fs = require('fs');

let getLink = (sdk_link) => {
  try {  
      var data = fs.readFileSync(sdk_link, 'utf8');
      console.log(data.toString());  
      return data  
  } catch(e) {
      console.log('Error:', e.stack);
  }
}

let download = async (uri, filename) => {
    let command = `curl -o ${filename} ${uri}`;
    let result = cp.execSync(command);
};

var sdk_link = "./sdk_link.txt"
var output = "./sdk.zip"

async function start() {
    await download( getLink(sdk_link), output )
}

start()
 ```
 Run
 ```
node download
```

**Reply by Armaldio** [reply]
Yeah, that's the goal: download directly from steam

But without authentication, I'm unable to download the file

**Reply by insraq** [reply]
> download directly from steam

You cannot. You need to be a "partner" to do that. So manual update is the only way.

**Reply by insraq** [reply]
@Armaldio Do you have access to Steamworks 1.53a?

**Reply by Armaldio** [reply]
Yes, this is the one I'm using in the CI currently

**Reply by insraq** [reply]
> Yes, this is the one I'm using in the CI currently

Great. I think your current solution should be okay. We cannot provide the SDK directly in the repo due to Steam NDA. Although Facepunch seems to have the C++ header files and DLLs in the repo https://github.com/Facepunch/Facepunch.Steamworks/tree/master/Generator/steam_sdk 🤷‍♂️

**Reply by Armaldio** [reply]
Currently I have a password protected zip of the steamworks sdk with only the public and redistributable_bin folders inside.
Is that OK or should I host myself the zip somewhere and download it on demand ?

**Reply by sleepingpandagames** [reply]
Hey there! @Armaldio @insraq I'm a Steam partner. Do you want me to ask directly to Steam support if it would be possible to host a zip file of their SDK in a public repo? Password protected or not. Or if they could suggest a solution. Maybe they can provide a secure link or something. I don't know how to code, but if I can help you push this forward in some ways then I'd love to.

**Reply by Armaldio** [reply]
Yes, that would be great 👍

**Comment by ceifa** [comment]
The steamworks SDK is downloadable from steamcmd using an anonymous authentication: https://steamdb.info/app/1007/

I think we can use it.

**Reply by Armaldio** [reply]
Ah !
Great, I was looking into steamcmd but indeed missed this one.

Looks promising, I will try that

**Reply by Armaldio** [reply]
Unfortunately, files seems not to be the ones we need.
Even more, the osx repo is empty

https://steamdb.info/depot/1004/
https://steamdb.info/depot/1005/
https://steamdb.info/depot/1006/

**Reply by sleepingpandagames** [reply]
That's what I was afraid of.. This repo not being up to date or the right files..
Btw still no answers on the community forum and from Valve itself.
